src/content/content.js

```diff
@@ -24,14 +24,28 @@ function planTheSwitcheroo (tabs) {
         // the entire selected mechanism is absolute shit. but idgaf.
         selectedEl, disruptedSelection;
 
+    var exposedContainer = document.createElement('div'),
+        shadowedContainer = exposedContainer.createShadowRoot();
+
+    exposedContainer.classList.add('switcheroo-container');
+
     filterList();
-    document.body.appendChild(container);
+    document.body.appendChild(exposedContainer);
+    // this is all inside our shadow world now
+
+    createStyleElement().then(function (style) {
+        shadowedContainer.appendChild(style);
+        // the following MUST happen after the shadow dom contains our style
+        // else we'd face a reflow, and worse, can cause autoscrolling of
+        // the page twice
+        shadowedContainer.appendChild(container);
```

Comment:
Bit of clarification: if the stylesheet is not already present in our (shadow) dom, and we create the tab list there, it would be appended to the dom like any other dom element, and the `scrollintoviewifneeded` call down there would trigger a scroll of the page, since without css, the container would just sit at the end of the document.

having the styles ready at first ensures we don't face the reflow, and don't cause an auto scroll.